### PR TITLE
Add--disable-historical-detection flag to slasher

### PR DESCRIPTION
**Pull request by 0xKiwi**
**What type of PR is this?**
Feature gating behind a flag

**What does this PR do? Why is it needed?**
This PR adds a new `--disable-historical-detection` flag so the slasher can run without reliance on knowing prior history or a new state management supported beacon node.

### CODE REVIEW

File: slasher/detection/service.go [hunk 1]
```diff
@@ -3,6 +3,8 @@ package detection
 import (
 	"context"
 
+	"github.com/prysmaticlabs/prysm/shared/featureconfig"
+
```
Comment (terencechain) [inline slasher/detection/service.go:7]:
```suggestion
```

Comment (0xKiwi) [reply slasher/detection/service.go:7]: Fixed, thanks!

File: slasher/usage.go [hunk 2]
```diff
@@ -5,6 +5,8 @@ import (
 	"io"
 	"sort"
 
+	"github.com/prysmaticlabs/prysm/shared/featureconfig"
```
Comment (rauljordan) [inline slasher/usage.go:8]:
```suggestion
```

**Comment by rauljordan** [comment]
Build fails @0xKiwi

File: slasher/usage.go [hunk 3]
```diff
@@ -5,7 +5,6 @@ import (
 	"io"
 	"sort"
 
-	"github.com/prysmaticlabs/prysm/shared/featureconfig"
```
Comment (0xKiwi) [inline slasher/usage.go]: lol what happened here

Comment (rauljordan) [reply slasher/usage.go]: My bad @0xKiwi apologize for the bad commit

Comment (0xKiwi) [reply slasher/usage.go]: No worries @rauljordan ! Was funny haha

**Pull request merged**